src/app/reporting/Engine.cpp

```diff
@@ -118,7 +119,8 @@ CHIP_ERROR Engine::BuildSingleReportDataAttributeReportIBs(ReportDataMessage::Bu
             ConcreteReadAttributePath pathForRetrieval(readPath);
             // Load the saved state from previous encoding session for chunking of one single attribute (list chunking).
             AttributeValueEncoder::AttributeEncodeState encodeState = apReadHandler->GetAttributeEncodeState();
-            err = RetrieveClusterData(apReadHandler->GetSubjectDescriptor(), attributeReportIBs, pathForRetrieval, &encodeState);
+            err = RetrieveClusterData(apReadHandler->GetSubjectDescriptor(), apReadHandler->IsFabricFiltered(), attributeReportIBs,
```

Comment:
Raising the question of whether RetrieveClusterData should just be handed the ReadHandler....

Reply:
I'm also having this question, however, while writing tests for fabric filtered read, I found we cannot have a real e2e test for attribute access interface (https://github.com/project-chip/connectedhomeip/issues/13486, and the TODOs in the unit test commit)

I would like to make this a part of larger refactor (infact, moving the code around) of the related code.

Reply:
Sure, I was not asking you to make this change in this PR; it's followup fodder.